src/nnet3/nnet-test-utils.cc

```diff
@@ -804,6 +804,24 @@ static void GenerateRandomComponentConfig(std::string *component_type,
       os << "input-dim=" << input_dim << " output-dim=" << output_dim;
       break;
     }
+    case 17: {
+      *component_type = "Convolutional1dComponent";
+      int32 patch_stride = 10 + Rand() % 50, patch_step = 1 + Rand() % 4, 
+	    patch_dim = 4 + Rand () % 5;
+
+      // decrease patch_stride until (patch_stride - patch_dim) % patch_step == 0
+      while ((patch_stride - patch_dim) % patch_step != 0)
```

Comment:
OK.  Please rebase your changes so that only the initial commit remains.
Dan


On Sun, Sep 20, 2015 at 5:49 PM, Yiming Wang <notifications@github.com>
wrote:

> In src/nnet3/nnet-test-utils.cc
> <https://github.com/kaldi-asr/kaldi/pull/156#discussion_r39935212>:
>
> > @@ -804,6 +804,24 @@ static void GenerateRandomComponentConfig(std::string *component_type,
> >        os << "input-dim=" << input_dim << " output-dim=" << output_dim;
> >        break;
> >      }
> > +    case 17: {
> > +      *component_type = "Convolutional1dComponent";
> > +      int32 patch_stride = 10 + Rand() % 50, patch_step = 1 + Rand() % 4,
> > +	    patch_dim = 4 + Rand () % 5;
> > +
> > +      // decrease patch_stride until (patch_stride - patch_dim) % patch_step == 0
> > +      while ((patch_stride - patch_dim) % patch_step != 0)
>
> Restored.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/156/files#r39935212>.
>